### PR TITLE
Fix invisible num colors, make buildable on NixOS

### LINKED ISSUE #14: Adjust line number color for light themes
The line numbering currently is not/barely visible with light themes.

### PR DESCRIPTION
**Pull request by gilice**
This fixes #14.
`flake.*` and `.envrc` help to build the application on the [NixOS linux distro](https://nixos.org)

Review: @dhzdhd

### CODE REVIEW

File: lib/src/home/views/home_view.dart [hunk 1]
```diff
@@ -5,13 +5,12 @@ import 'package:flutter/material.dart';
...
 
 class HomeView extends ConsumerWidget {
-  const HomeView({Key? key}) : super(key: key);
   static const routeName = '/';
+  const HomeView({Key? key}) : super(key: key);
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
     return const Scaffold(
-      resizeToAvoidBottomInset: true,
```
Comment (gilice) [inline lib/src/home/views/home_view.dart:14]: I removed this because it already defaults to true

File: lib/src/home/widgets/code_field.dart [hunk 2]
```diff
@@ -41,101 +41,6 @@ class CodeFieldWidgetState extends State<CodeFieldWidget> {
   String longestLine = '';
   int _selectedLine = 0;
 
-  @override
```
Comment (gilice) [inline lib/src/home/widgets/code_field.dart:44]: Those are autoformat changes, sorry.

File: lib/src/home/widgets/code_field.dart [hunk 3]
```diff
@@ -230,4 +136,110 @@ class CodeFieldWidgetState extends State<CodeFieldWidget> {
...
+    }
+    return count;
+  }
+
+  Color? _getLineNumberColor(
+      bool selected, Map<String, TextStyle>? theme, String? rootKey) {
+    if (theme != null && rootKey != null) {
+      return selected
+          ? theme[rootKey]!.color
+          : theme[rootKey]!.color?.withAlpha(128);
```
Comment (gilice) [inline lib/src/home/widgets/code_field.dart:191]: Those look the most sensible to me.
I didn't check if theme can ever be null here - I think it shouldn't, but the previous version is still included as a fallback.

File: pubspec.yaml [hunk 4]
```diff
@@ -7,7 +7,7 @@ version: 0.3.0
 
 environment:
   sdk: ">=2.17.0 <3.0.0"
-  flutter: ^3.3.0
+  flutter: ^3.0.0
```
Comment (gilice) [inline pubspec.yaml:10]: NixOs didn't package latest flutter yet, so I couldn't build the project. Feel free to remove this, if you want.

File: default.nix [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# Desc: Default.nix with flake-compat
```
Comment (gilice) [inline default.nix:1]: This too, is a helper file for nixOs packaging, which I'll do soon (so this app will be in the official NixOs repo `nixpkgs`)

**Comment by gilice** [comment]
I also made the minimum flutter version a bit lower, because flutter 3.3 isn't packaged for my Linux distro yet.

@dhzdhd do you want me to revert that?

**Comment by dhzdhd** [comment]
> I also made the minimum flutter version a bit lower, because flutter 3.3 isn't packaged for my Linux distro yet.
> 
> @dhzdhd do you want me to revert that?

Yes, I need v3.3
Also, please make an issue for support for building on nixOS. Let this PR be only about fixing the colors.

**Comment by gilice** [comment]
Hey @dhzdhd !
I did as you requested and updated the branch; now the only thing this does is fixing the invisible num colors

**Review by dhzdhd — APPROVED** [review]
LGTM!

**Pull request merged**